app/assets/javascripts/discourse/components/radio-button.js.es6

```diff
@@ -14,10 +14,15 @@ export default Component.extend({
 
   click() {
     const value = $(this.element).val();
-    if (this.selection === value) {
-      this.set("selection", undefined);
+
+    if (this.onChange) {
+      this.onChange(value);
```

Comment:
I have the same change on a branch in local currently 😅 

nice

Reply:
A stopgap before `{{on}}` element modifier arrives to Discourse codebase. 😉